### PR TITLE
feature(Set dynamic Base Index growth rate)

### LINKED ISSUE #6005: Enter the annual rate of increase of the configurable basic vanga indices
During the implementation of the basic indices of Vanga, the rate of increase had been fixed whereas this value should be variable and configurable, it is therefore a question of defining this value as a parameter of the company

### PR DESCRIPTION
**Pull request by lomamech**
- Manual entry of the rate of increase of the base index
- Improvement of the UpdateStaffingIndices () procedure
   by allowing only the selection of employees defined for the period
- Rounding of basic index calculation
closes #6005

### CODE REVIEW

**Comment by lomamech** [comment]
@mbayopanda @jmcameron I need a review for this PR

File: client/src/i18n/en/form.json [hunk 1]
```diff
@@ -329,6 +329,7 @@
             "BALANCE_SHEET": "Balance Sheet",
             "BANK_ACCOUNT": "Bank Account",
             "BANK": "Bank",
+            "BASE_INDEX_GROWTH_RATE": "Base index growth rate",
```
Comment (jmcameron) [inline client/src/i18n/en/form.json:333]: What are the units for this?   %/year?

File: client/src/modules/enterprises/enterprises.html [hunk 2]
```diff
@@ -291,6 +291,18 @@
                 on-change-callback="EnterpriseCtrl.enableIndexPaymentSetting(value)">
               </bh-yes-no-radios>
 
+              <div class="form-group" ng-class="{ 'has-error' : EnterpriseForm.$submitted && EnterpriseForm.base_index_growth_rate.$invalid }">
+                <label class="control-label" translate>
+                  FORM.LABELS.BASE_INDEX_GROWTH_RATE
+                </label>
```
Comment (jmcameron) [inline client/src/modules/enterprises/enterprises.html:297]: I would really like to see a description of what this option is how it is used in the system (a _HELP_TEXT msg) like on the previous settings.

**Review by jmcameron — COMMENTED** [review]
As I indicated in the code, I would like the enterprises settings page to have an exaplanation of what this new setting is and how it is used.    It is not obvious to me that this new setting is being used anywhere currently (must have missed it in the code).

I tested this with bhima_test and the user interface works for setting the value.

File: client/src/i18n/en/form.json (hunk 1 above)
Comment (lomamech) [reply client/src/i18n/en/form.json:333]: Yes

Comment (lomamech) [reply client/src/i18n/en/form.json:333]: Yes

Comment (jmcameron) [reply client/src/i18n/en/form.json:333]: I suggest you include that in the label.  It see in the gui it has '%'.  Could this be update to be '%/year' or '%/an' with language tokens.

Comment (lomamech) [reply client/src/i18n/en/form.json:333]: Okay :white_check_mark:

File: client/src/modules/enterprises/enterprises.html (hunk 2 above)
Comment (lomamech) [reply client/src/modules/enterprises/enterprises.html:297]: :white_check_mark:

**Review by jniles — COMMENTED** [review]
@lomamech, this PR looks good to me.  As @jmcameron noted, having the units would be nice - it isn't clear to me that the index grows annually (instead of monthly, etc).

At some point, we might want to make a "Payroll Settings" page, like we do for "Stock Settings".  That shouldn't be in this PR though.

**Comment by lomamech** [comment]
Usually it's after each year, it's sort of a bonus for one year of seniority

**Comment by jniles** [comment]
@lomamech right - add that to the text.  A user will never know that otherwise.

**Comment by lomamech** [comment]
@jniles @jmcameron I think I ended up with this pull request, in my next request request I think adding the option Payroll Setting and I will need some elements of this Pull Request

**Comment by lomamech** [comment]
@jniles @jmcameron I think I ended up with this pull request, in my next request request I think adding the option Payroll Setting and I will need some elements of this Pull Request

**Comment by lomamech** [comment]
@jniles @jmcameron I think I ended up with this pull request, in my next request request I think adding the option Payroll Setting and I will need some elements of this Pull Request

**Comment by jmcameron** [comment]
> @jniles @jmcameron I think I ended up with this pull request, in my next request request I think adding the option Payroll Setting and I will need some elements of this Pull Request

Sounds good.  merge your new PR with the old PR and when the new PR is successful close the old one.  In the meantime, I suggest marking your old PR as "do not merge".

**Review by jniles — APPROVED** [review]
LGTM.

bors r+

**Pull request merged**